### PR TITLE
Assorted CMake Fixes to Improve Submodule Usability

### PR DESCRIPTION
**Pull request by cwfitzgerald**
There's a lot going on here, but I tried to break it down into small logical commits for easy review.

https://github.com/NovaMods/profiler/pull/1 must be resolved first.

Any questions, hmu.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -198,26 +175,50 @@ set_target_properties(nova-renderer PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
...
+	                 spirv-cross-hlsl
+	                 spirv-cross-reflect
+	                 spirv-cross-util
+	                 nova-profiler
+	                 miniz
+	                 glslang
+	                 SPIRV
+	                 ${SPIRV_SHARED_LIBRARIES}
+	                 ${Vulkan_LIBRARIES}
+	                 ${GLFW_LIBRARY_NAME})
```
Comment (DethRaid) [inline CMakeLists.txt]: We don't use GLFW. I guess it shuck into the original CMake file but it should be removed, especially since you removed the GLFW submodule

File: CMakeLists.txt [hunk 2]
```diff
@@ -198,26 +175,50 @@ set_target_properties(nova-renderer PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
...
     set(COMMON_LINK_LIBS ${COMMON_LINK_LIBS} d3d12 dxgi d3dcompiler)
-    target_include_directories(nova-renderer PRIVATE
-            "C:\\Program Files (x86)\\Windows Kits\\10\\Include\\10.0.17134.0\\shared"
-            "C:\\Program Files (x86)\\Windows Kits\\10\\Include\\10.0.17134.0\\um")
+    target_include_directories(nova-renderer PRIVATE ${WinSDK_DIRS})
     target_compile_definitions(nova-renderer PUBLIC _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING NOMINMAX)
 
-    # Visual Studio is stupid. Is it really that hard to use filesystem directories to organize code like literally
-    # everyone else?
+	# Visual Studio is stupid. Is it really that hard to use filesystem directories to organize code like literally
```
Comment (DethRaid) [inline CMakeLists.txt]: Why'd the indentation change?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
-if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    set(NOVA_CXX_FLAGS "${NOVA_CXX_FLAGS} -pthread -g -rdynamic")
-endif()
+find_package(Threads REQUIRED)
 
 # End-to-end test
-add_executable(nova-test-end-to-end src/end_to_end_runner.cpp src/general_test_setup.hpp)
-add_dependencies(nova-test-end-to-end nova-renderer)
-target_include_directories(nova-test-end-to-end PRIVATE ${CMAKE_CURRENT_LIST_DIR}/3rdparty/googletest/googletest/include)
+add_executable(nova-test-end-to-end ${CMAKE_CURRENT_SOURCE_DIR}/src/end_to_end_runner.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/general_test_setup.hpp)
```
Comment (DethRaid) [inline tests/CMakeLists.txt]: Why the change to the paths to the test files?

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -1,33 +1,29 @@
...
 target_compile_definitions(nova-test-end-to-end PRIVATE CMAKE_DEFINED_RESOURCES_PREFIX="${CMAKE_CURRENT_LIST_DIR}/resources/")
-target_link_libraries(nova-test-end-to-end nova-renderer gtest gtest_main)
+target_link_libraries(nova-test-end-to-end PRIVATE nova-renderer gtest gtest_main Threads::Threads)
 
 add_test(NAME nova-end-to-end-test COMMAND nova-test-end-to-end)
 
 # Unit tests
 add_executable(nova-test-unit unit_tests/loading/filesystem_test.cpp src/general_test_setup.hpp unit_tests/loading/shaderpack/shaderpack_validator_tests.cpp)
-add_dependencies(nova-test-unit nova-renderer)
-target_include_directories(nova-test-unit PRIVATE ${CMAKE_CURRENT_LIST_DIR}/3rdparty/googletest/googletest/include)
```
Comment (DethRaid) [inline tests/CMakeLists.txt:26]: Why don't we need to specify the GoogleTest include directory?

**Review by DethRaid — CHANGES_REQUESTED** [review]

File: tests/CMakeLists.txt (hunk 3 above)
Comment (cwfitzgerald) [reply tests/CMakeLists.txt]: Because I was compiling c++ as c and I couldn't figure out why so I was trying everything I could to try to get cmake to figure out it was c++... will fix

**Review by DethRaid — APPROVED** [review]
